### ISSUE TITLE
Build failure: amdvlk

### ISSUE THREAD

**Issue by miyoku157** [NixOS/nixpkgs#349362]
### Steps To Reproduce

1. nix flake update
2. sudo nixos-rebuild switch --flake .

### Build log

```
error: builder for '/nix/store/h2adl88hvwrndpq24kw8pav8hwzia1yb-amdvlk-2024.Q3.2.drv' failed with exit code 1;
       last 10 log lines:
       > /build/amdvlk-src/drivers/pal/src/util/lnx/lnxSysUtil.cpp:961:80: warning: narrowing conversion of '(((long unsigned int)fileNameSize) - 1)' from 'long unsigned int' to 'Util::uint32' {aka 'unsigned int'} [-Wnarrowing]
       >   961 |             fileNames[fileIndex] = StringView<char>{ temp.Data(), fileNameSize - 1 };
       >       |                                                                   ~~~~~~~~~~~~~^~~
       > [337/2845] Building CXX object pal/CMakeFiles/palUtil.dir/src/util/jsonWriter.cpp.o
       > [338/2845] Building CXX object pal/CMakeFiles/palUtil.dir/src/util/lz4Compressor.cpp.o
       > [339/2845] Building CXX object pal/CMakeFiles/palUtil.dir/src/util/conditionVariable.cpp.o
       > [340/2845] Building CXX object pal/CMakeFiles/palUtil.dir/src/util/dbgLogger.cpp.o
       > [341/2845] Building CXX object pal/CMakeFiles/palUtil.dir/src/util/file.cpp.o
       > [342/2845] Building CXX object compiler/llpc/llvm/lib/DebugInfo/CodeView/CMakeFiles/LLVMDebugInfoCodeView.dir/EnumTables.cpp.o
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/h2adl88hvwrndpq24kw8pav8hwzia1yb-amdvlk-2024.Q3.2.drv'.
error: 1 dependencies of derivation '/nix/store/95g2hc7jagj66vv645d3bfng2n9hdsj7-graphics-drivers.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5f28bqlwpnqm94dfsvha3qa37lmh5ba8-steam-run-usr-target.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5zldpkz3rqg327synmdm5pmdp148l2q7-steam-usr-target.drv' failed to build
error: 1 dependencies of derivation '/nix/store/n9bavgghng1zqp1w80gc01wq7acm6i0h-graphics-driver.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/scg2qn7kpc7kh93ddpdh4r9g9vdagbji-steam-fhs.drv' failed to build
error: 1 dependencies of derivation '/nix/store/66lq1cfv1n8r7jwj60g8m4lgd8f6glv9-steam-run-fhs.drv' failed to build
error: 1 dependencies of derivation '/nix/store/0p9mi5gqkwlx2bd82yqqbnlgf4v3id0p-steam-bwrap.drv' failed to build
error: 1 dependencies of derivation '/nix/store/lcy3pb5r5ms44lsdrhy75lsskdm13p8v-steam-run-bwrap.drv' failed to build
error: 1 dependencies of derivation '/nix/store/06gq00a6dr1gpxwy3kdwlzrpvgdihlpg-tmpfiles.d.drv' failed to build
error: 1 dependencies of derivation '/nix/store/4hkln1z15i5i366c7kw3lqli65irbbi0-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/w5wrnq81zk5xgl3a5l2rdfz454pbskcv-steam-run.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rqn8invcs31jg95rnd0hdx5in2mjmaqg-steam.drv' failed to build
error: 1 dependencies of derivation '/nix/store/3iiysw7skis2ai0vjpfkflxmhmm3spni-nixos-system-nixoku157-24.11.20241014.a3c0b3b.drv' failed to build

nixos log:
https://gist.github.com/miyoku157/94b67129be29ddbb893882d78ee196ba
```

### Notify maintainers

@Flakebi
@JohnRTitor 

### Metadata

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.9-zen1, NixOS, 24.11 (Vicuna), 24.11.20240926.1925c60`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.7`
 - channels(root): `"nixos-23.11"`
 - nixpkgs: `/nix/store/fpivx4sjcp2vk4rp9nhliln5cwcp3kc6-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Thiago-Assis-T** [comment]
Just had the same issue...

**Comment by x10an14** [comment]
I've now got a similar issue, but not exactly the same:
```
[1822/2845] Building CXX object compiler/llpc/llvm/lib/CodeGen/SelectionDAG/CMakeFiles/LLVMSelectionDAG.dir/InstrEmitter.cpp.o
[1823/2845] Building CXX object compiler/llpc/llvm/lib/CodeGen/SelectionDAG/CMakeFiles/LLVMSelectionDAG.dir/FunctionLoweringInfo.cpp.o
[1824/2845] Building CXX object compiler/llpc/llvm/lib/CodeGen/SelectionDAG/CMakeFiles/LLVMSelectionDAG.dir/FastISel.cpp.o
[1825/2845] Building CXX object compiler/llpc/llvm/lib/CodeGen/SelectionDAG/CMakeFiles/LLVMSelectionDAG.dir/LegalizeDAG.cpp.o
[1826/2845] Building AMDGPUGenAsmMatcher.inc...
[1827/2845] Building CXX object compiler/llpc/llvm/lib/CodeGen/SelectionDAG/CMakeFiles/LLVMSelectionDAG.dir/DAGCombiner.cpp.o
/build/amdvlk-src/drivers/llvm-project/llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp: In function 'llvm::SDValue FoldIntToFPToInt(llvm::SDNode*, llvm::SelectionDAG&)':
/build/amdvlk-src/drivers/llvm-project/llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp:17939:23: warning: possibly dangling reference to a temporary [-Wdangling-reference]
17939 |   const fltSemantics &Sem = N0.getValueType().getFltSemantics();
      |                       ^~~
/build/amdvlk-src/drivers/llvm-project/llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp:17939:62: note: the temporary was destroyed at the end of the full expression 'N0.llvm::SDValue::getValueType().llvm::EVT::getFltSemantics()'
17939 |   const fltSemantics &Sem = N0.getValueType().getFltSemantics();
      |                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
ninja: build stopped: subcommand failed.
```

This after upgrading my flake input:
```
• Updated input 'home-manager':
    'github:nix-community/home-manager/038630363e7de57c36c417fd2f5d7c14773403e4' (2024-10-07)
  → 'github:nix-community/home-manager/e78cbb20276f09c1802e62d2f77fc93ec32da268' (2024-10-17)
• Updated input 'nixpkgs':
    'github:nixos/nixpkgs/c31898adf5a8ed202ce5bea9f347b1c6871f32d1' (2024-10-06)
  → 'github:nixos/nixpkgs/a3c0b3b21515f74fd2665903d4ce6bc4dc81c77c' (2024-10-14)
```

Full error log: https://paste.sr.ht/~x10an14/533580fca03780e590ee0d180cf3eed498db0bf0

**Comment by liff** [comment]
Duplicate of #348903, fix in #349246.

**Comment by JohnRTitor** [comment]
Duplicate of #348903
